File: sdk/anthropic/python/quickstart.py
"""Claude Agent SDK — Python quickstart.

Setup:
    pip install -r requirements.txt
    export ANTHROPIC_API_KEY="sk-ant-..."

Run:
    python quickstart.py

Requires Python 3.10+. The SDK bundles Claude Code automatically.
Docs: https://github.com/anthropics/claude-agent-sdk-python
"""

import anyio

from claude_agent_sdk import (
    ClaudeAgentOptions,
    ClaudeSDKClient,
    query,
    tool,
    create_sdk_mcp_server,
)


# 1) Simplest possible call: stream messages from a one-shot query.
async def basic_query() -> None:
    async for message in query(prompt="What is 2 + 2? Answer in one word."):
        print(message)


# 2) A custom in-process tool exposed to the agent via an SDK MCP server.
@tool("add", "Add two numbers", {"a": float, "b": float})
async def add(args: dict) -> dict:
    result = args["a"] + args["b"]
    return {"content": [{"type": "text", "text": f"{result}"}]}


calculator = create_sdk_mcp_server(name="calculator", version="1.0.0", tools=[add])


# 3) Configure options: system prompt, allowed built-in tools, and our custom tool.
async def with_tools() -> None:
    options = ClaudeAgentOptions(
        system_prompt="You are a concise assistant. Use tools when they help.",
        allowed_tools=["Read", "Write", "Bash", "mcp__calculator__add"],
        mcp_servers={"calculator": calculator},
    )
    async with ClaudeSDKClient(options=options) as client:
        await client.query("Use the add tool to compute 17 + 25, then explain briefly.")
        async for msg in client.receive_response():
            print(msg)


async def main() -> None:
    print("=== basic_query ===")
    await basic_query()
    print("\n=== with_tools ===")
    await with_tools()


if __name__ == "__main__":
    anyio.run(main)

File: sdk/anthropic/typescript/quickstart.ts
/**
 * Claude Agent SDK — TypeScript quickstart.
 *
 * Setup:
 *   npm install
 *   export ANTHROPIC_API_KEY="sk-ant-..."
 *
 * Run:
 *   npx tsx quickstart.ts
 *
 * Docs: https://github.com/anthropics/claude-agent-sdk-typescript
 */

import { query } from "@anthropic-ai/claude-agent-sdk";

// 1) Simplest possible call: async-iterate the streamed messages.
async function basicQuery(): Promise<void> {
  for await (const message of query({
    prompt: "What is 2 + 2? Answer in one word.",
  })) {
    console.log(message);
  }
}

// 2) With options: a system prompt and an allow-list of built-in tools.
async function withOptions(): Promise<void> {
  for await (const message of query({
    prompt: "Create a file called hello.txt containing 'hi', then read it back.",
    options: {
      systemPrompt: "You are a concise coding assistant.",
      allowedTools: ["Read", "Write", "Bash"],
    },
  })) {
    console.log(message);
  }
}

async function main(): Promise<void> {
  console.log("=== basicQuery ===");
  await basicQuery();
  console.log("\n=== withOptions ===");
  await withOptions();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
